Source/Csla.Shared/Server/DataPortal.cs

```diff
@@ -62,9 +65,20 @@ protected DataPortal(string cslaAuthorizationProviderAppSettingName)
     protected DataPortal(Type authProviderType)
     {
       if (null == authProviderType)
-        throw new ArgumentNullException("authProviderType", Resources.CslaAuthenticationProviderNotSet);
+        throw new ArgumentNullException(nameof(authProviderType), Resources.CslaAuthenticationProviderNotSet);
       if (!typeof(IAuthorizeDataPortal).IsAssignableFrom(authProviderType))
```

Comment:
Curious....why does this constructor take a `Type`? Couldn't you use a constrained generic to ensure that the type is of a type `IAuthorizeDataPortal`?

Reply:
This is old code that works and I see no value in changing. The only reason for this at all is to enable some unit testing scenarios.